Gini-iOS-SDK/GINIDocumentTaskManager.m

```diff
@@ -185,55 +267,32 @@ - (BFTask *)privatePollDocumentWithId:(NSString *)documentId
             }];
             // Otherwise return the document.
         } else {
-            return [GINIDocument documentFromAPIResponse:polledDocument withDocumentManager:self];
+            return [GINIDocument documentFromAPIResponse:polledDocument];
         }
     }];
 }
 
-- (BFTask *)getPreviewForPage:(NSUInteger)page
-                   ofDocument:(GINIDocument *)document
-                     withSize:(GiniApiPreviewSize)size {
-    return [self getPreviewForPage:page ofDocument:document withSize:size cancellationToken:nil];
-}
-
-- (BFTask *)getPreviewForPage:(NSUInteger)page
-                   ofDocument:(GINIDocument *)document
-                     withSize:(GiniApiPreviewSize)size
-            cancellationToken:(BFCancellationToken *)cancellationToken {
-    NSParameterAssert(page > 0);
-    NSParameterAssert(page <= document.pageCount);
-    
-    BFTask *pageTask = [_apiManager getPreviewForPage:page ofDocument:document.documentId withSize:size cancellationToken:cancellationToken];
-    return GINIhandleHTTPerrors(pageTask);
-}
-
-#pragma mark - Extraction methods
-- (BFTask *)getExtractionsForDocument:(GINIDocument *)document {
-    return [self getExtractionsForDocument:document cancellationToken:nil];
-}
-
-- (BFTask *)getExtractionsForDocument:(GINIDocument *)document
-                    cancellationToken:(BFCancellationToken *)cancellationToken {
+- (BFTask *)updateDocument:(GINIDocument *)document
+        updatedExtractions:(NSDictionary *)updatedExtractions
+         cancellationToken:(BFCancellationToken *)cancellationToken {
     NSParameterAssert([document isKindOfClass:[GINIDocument class]]);
     
-    BFTask *extractionsTask = [self createExtractionsForGetTask:[_apiManager getExtractionsForDocument:document.documentId
-                                                                                     cancellationToken:cancellationToken]];
-    return GINIhandleHTTPerrors(extractionsTask);
-}
-
-- (BFTask *)getIncubatorExtractionsForDocument:(GINIDocument *)document {
-    return [self getIncubatorExtractionsForDocument:document cancellationToken:nil];
-}
-
-- (BFTask *)getIncubatorExtractionsForDocument:(GINIDocument *)document
-                             cancellationToken:(BFCancellationToken *)cancellationToken {
-    NSParameterAssert([document isKindOfClass:[GINIDocument class]]);
+    NSMutableDictionary *filteredUpdatedExtractions = [NSMutableDictionary new];
     
-    BFTask *extractionsTask = [self createExtractionsForGetTask:[_apiManager getIncubatorExtractionsForDocument:document.documentId
-                                                                                              cancellationToken:cancellationToken]];
-    return GINIhandleHTTPerrors(extractionsTask);
+    NSArray *keys = @[@"paymentReference", @"iban", @"bic", @"amountToPay", @"paymentRecipient", @"paymentPurpose"];
```

Comment:
This is how you work with feedback, right? I know e only get feedback for these specific fields, but for someone without the domain knowledge it would be difficult to understand this code. Maybe you can add a comment and possibly extract these fields somewhere?